main.go

```diff
@@ -249,25 +250,7 @@ func main() {
 
 	// Register grpc prometheus server
 	grpc_prometheus.Register(grpcServer)
-	http.Handle("/metrics", promhttp.Handler())
-	http.HandleFunc("/_packet/healthcheck", healthCheckHandler)
-	http.HandleFunc("/_packet/version", versionHandler)
-	http.HandleFunc("/2009-04-04", ec2Handler) // workaround for making trailing slash optional
-	http.HandleFunc("/2009-04-04/", ec2Handler)
-
-	err = registerCustomEndpoints()
-	if err != nil {
-		logger.Fatal(err, "could not register custom endpoints")
-	}
-
-	logger.With("port", *metricsPort).Info("Starting http server")
-	go func() {
-		err := http.ListenAndServe(fmt.Sprintf(":%d", *metricsPort), nil)
-		if err != nil {
-			logger.Error(err, "failed to serve http")
-			panic(err)
-		}
-	}()
+	ServeHTTP()
```

Comment:
we should call `ParseTrustedProxies` and pass it to `ServeHTTP` here and also use it to pass to `GRPCMiddlewares` above.